### PR TITLE
Chapter 7: Sign up

### PR DESCRIPTION
**Pull request by nguyenquanganhc**
Sign up form
image
https://docs.google.com/document/d/1PtAhV9qBB9Aviz20GzsYreERiioRS8MMPJh1mhhroDI/edit?usp=sharing

### CODE REVIEW

**Comment by nguyenquanganhc** [comment]
ready

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,3 +1,31 @@
 class UsersController < ApplicationController
-  def new; end
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb]:
```suggestion
    @user = User.new user_params
```

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+    if @user.save
+      flash[:success] = "Welcome to the Sample App!"
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb]: thêm i18n

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,3 +1,31 @@
...
+      redirect_to @user
+    else
+      render action: "new"
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb]:
```suggestion
      render :new
```

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,3 +1,31 @@
...
+    end
+  end
+
+  def show
+    @user = User.find_by id: params[:id]
+    if @user.blank?
+      flash[:error] = "Could not found user"
+      redirect_to(root_path)
+    else
+      render action: "show"
```
Comment (ngocvt-0484) [inline app/controllers/users_controller.rb]: fix

Comment (ngocvt-0484) [reply app/controllers/users_controller.rb]: thêm flash message nữa

File: app/helpers/users_helper.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+module UsersHelper
+  # Returns the Gravatar for the given user.
+  def gravatar_for user, size: 80
```
Comment (ngocvt-0484) [inline app/helpers/users_helper.rb]: cho 80 vài setting

File: app/views/users/new.html.erb [hunk 6]
```diff
@@ -1,6 +1,27 @@
...
+<h1>
+  <%= t ".fore_title" %>
+</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@user) do |f| %>
+      <%= render "shared/error_messages" %>
+      
+      <%= f.label :name, t(".name") %>
+      <%= f.text_field :name, class: 'form-control' %>
```
Comment (ngocvt-0484) [inline app/views/users/new.html.erb]: dùng nháy "

**Comment by nguyenquanganhc** [comment]
ready

File: app/views/users/new.html.erb [hunk 7]
```diff
@@ -1,6 +1,27 @@
...
+      <%= f.password_field :password, class: "form-control" %>
+      
+      <%= f.label :password_confirmation, t(".pass_confirm") %>
+      <%= f.password_field :password_confirmation, class: "form-control" %>
+      
+      <%= f.submit t(".submit"), class: "btn btn-primary" %>
+    <% end %>
+  </div>
 </div>
+
```
Comment (ngocvt-0484) [inline app/views/users/new.html.erb]: thừa line

File: app/views/shared/_error_messages.html.erb [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(@user.errors.count, "error") %>.
```
Comment (ngocvt-0484) [inline app/views/shared/_error_messages.html.erb]: i18n

**Comment by nguyenquanganhc** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**